aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputStream.java

```diff
@@ -218,41 +254,49 @@ private void uploadParts() {
 
     stagingFiles.stream()
         // do not upload the file currently being written
-        .filter(f -> closed || !f.equals(currentStagingFile))
+        .filter(f -> closed || !f.file().equals(currentStagingFile))
         // do not upload any files that have already been processed
-        .filter(Predicates.not(multiPartMap::containsKey))
+        .filter(Predicates.not(f -> multiPartMap.containsKey(f.file())))
         .forEach(f -> {
           UploadPartRequest.Builder requestBuilder = UploadPartRequest.builder()
               .bucket(location.bucket())
               .key(location.key())
               .uploadId(multipartUploadId)
               .partNumber(stagingFiles.indexOf(f) + 1)
-              .contentLength(f.length());
+              .contentLength(f.file().length());
```

Comment:
Within this block, there would be a lot fewer changes if you left `f` unchanged: `.foreach(fileAndDigest -> { File f = fileAndDigest.file(); ... })`.